src/Actions/Order/UpdateLockedOrder.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace FluxErp\Actions\Order;
+
+use FluxErp\Actions\FluxAction;
+use FluxErp\Models\Order;
+use FluxErp\Rulesets\Order\UpdateLockedOrderRuleset;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Arr;
+
+class UpdateLockedOrder extends FluxAction
+{
+    protected function boot(array $data): void
+    {
+        parent::boot($data);
+        $this->rules = resolve_static(UpdateLockedOrderRuleset::class, 'getRules');
+    }
+
+    public static function description(): ?string
+    {
+        return 'Update an order regardless of its locked state.';
+    }
+
+    public static function models(): array
+    {
+        return [Order::class];
+    }
+
+    public function performAction(): Model
+    {
+        $addresses = Arr::pull($this->data, 'addresses', []);
```

Comment:
remove [] as fallback, as it removes all attached addresses if intentional not having the 'addresses' key.